frontend/server/src/DAO/Contests.php

```diff
@@ -399,7 +400,7 @@ final public static function getAllContestsOwnedByUser(
     /**
      * Returns all contests where a user is participating in.
      *
-     * @return list<Contest>
+     * @return list<Contestv2>
```

Comment:
```suggestion
     * @return list<Contest>
```

aquí no hay necesidad de declarar un tipo nuevo: se puede modificar el anterior. a final de cuentas, `getContests()` va a regresar esto.

Reply:
Con el penúltimo cambio se vio que si era necesario crear un PSALM Type para **`Contestv2`**, el motivo de esto es que existen métodos en el DAO que usan ese parámetro pero que no forman parte del método **`getContestList()`** del controlador que es utilizado para el listado de los concursos, por lo que no es de vital importancia para esas Queries tener la columna "**No. de Concursantes**" o "**Contestants**", de hecho esas queries no las modifiqué porque no aplican para este proyecto.

Reply:
si esto no fuera a reemplazar `Contest`, entonces no se podría llamar `Contestv2` porque en realidad no sería una versión 2 sino algo aparte que va a coexistir.

habiendo dicho eso, creo que es mejor tener un sólo tipo `Contest`. se pueden modificar las otras consultas para obtener la misma información? así no hay discrepancia entre las APIs.

Reply:
De hecho si lo va a reemplazar.

Sobre lo otro, eso no forma parte del propósito de este PR no?

Reply:
Ya que las otras Queries no se usan para el listado de concursos, tienen otros propósitos y otras columnas, lo revisé cuidadosamente y no le veo sentido hacer eso que comentas.

Reply:
ya no entendí: sí lo va a reemplazar? (eso implica que el 100% de los lugares donde se usa `Contest` hoy en día van a usar las nuevas columnas) pero no vas a ajustar los otros lugares donde se usa `Contest`. eso suena contradictorio.

Reply:
sobre el "no ver sentido" de hacerlo, estaría cool que en el futuro podamos desplegar _todos_ los lugares donde se muestran los concursos usando los mismos componentes, así que hay que traer la misma información. puedes hacer ese cambio porfa? lo que sí se vale es hacerlo en un cambio aparte.

Reply:
Pff tienes razón, pero es que sabes que pasa, que en el archivo del DAO hay varios métodos que tienen un valor de retorno que no depende de un PSALM Type en específico, sin embargo, es prácticamente el mismo al de **`Contest`** PSALM Type, entonces esas inconsistencias me confundieron al inicio (de hecho hasta el momento estoy confundido por eso jeje), me hace pensar que hay consultas que tienen propósitos diferentes con columnas diferentes.

Reply:
De hecho son sólo 5 métodos los que tienen como retorno `list<Contest>` ( y de esos sólo uno es el único que forma parte de la función del controlador `getContestList()` que es **`getContestsParticipating()`** ), de ahí en fuera, `getAllContestsForIdentity()` (por ejemplo) no tiene ese PSALM Type, a pesar de que también forma parte de `getContestList()`.

Reply:
Y si, claro, puedo hacer ese cambio, pero cómo crees que sería mejor en este PR o en otro a parte? Yo pienso que sería mejor en otro para no atrasar este PR con el tema de revisión, ya que el merge de los PRs para agregar las nuevas 3 columnas es algo que requiero pronto para poder realizar el PR de los Cards (donde le doy diseño a los concursos en la interfaz gráfica), sólo para que lo contemples, sin embargo, estoy abierto a cualquier opción.

Reply:
> en el archivo del DAO hay varios métodos que tienen un valor de retorno que no depende de un Psalm Type en específico, sin embargo, es prácticamente el mismo al de Contest Psalm Type

welp, eso está mal y hay que corregirlo luego.

> puedo hacer ese cambio, pero cómo crees que sería mejor? en este PR o en otro a parte?

arriba en este mismo thread menciono que se vale hacerlo en un cambio aparte. con este acuerdo entonces sí se vale dejarlo como `Contestv2` porque ahora sí va a reemplazarlo pronto.

Reply:
Ok, vale, abro el Issue en un momento más.

Sería entonces un Issue/PR por cada columna nueva verdad ¿? Ya que actualmente tenemos 3 PRs para las 3 columnas que serán agregadas.

Reply:
para esos se vale hacer un solo PR para todos porque el cambio va a ser bastante más peque.

Reply:
BTW, alguna sugerencia para el nombre de esos Issues/PRs? tipo "Unificando Contest y Contestv2" jeje ¿?

Reply:
seeeh, el nombre no es muy importante, y ese suena razonable.

Reply:
Ok, vale :)